### PR TITLE
feat(debug-options): add request and response interceptors

### PR DESCRIPTION
**Pull request by raminjafary**
Hey :)
This PR is to add a few functionalities by allowing intercepting requests and responses with a similar experience to `axios` module. The following methods are added:

- `onRequest`
- `onResponse`
- `onResponseError`
- `onRequestError`
- `onError`
- `installDebugInterceptors`. This might be used as a simple plugin to log all responses and errors, if any (example included).

### CODE REVIEW

File: lib/core/HTTP.ts [hunk 1]
```diff
@@ -121,4 +121,53 @@ export default class HTTP {
...
+    this.onRequestError((error: AxiosError) => {
+      console.info('Request error:', error)
+    })
+
+    this.onResponseError((error: AxiosError) => {
+      console.info('Response error:', error)
+    })
+
+    this.onResponse((response: AxiosResponse) => {
+      console.info(
```
Comment (farnabaz) [inline lib/core/HTTP.ts:164]: We should avoid using `console`, [Cosnola](https://github.com/nuxt-contrib/consola) would be a great replacement

File: example/nuxt.config.js [hunk 2]
```diff
@@ -15,6 +15,10 @@ export default {
       }
     }
   },
+  // Nuxt.js plugins
+  plugins: [
```
Comment (farnabaz) [inline example/nuxt.config.js:19]: I was thinking of creating an internal plugin to enable the debuging process without need any custom plugins.
We can have `debug` option inside module options and register a our debug plugin if it is `true`
What do you think about it?

**Review by farnabaz — CHANGES_REQUESTED** [review]

File: example/nuxt.config.js (hunk 2 above)
Comment (raminjafary) [reply example/nuxt.config.js:19]: I do totally agree. I'll update my PR with this change included!

File: lib/core/HTTP.ts (hunk 1 above)
Comment (raminjafary) [reply lib/core/HTTP.ts]: Right, I'll do that!

**Comment by raminjafary** [comment]
I updated my PR.

**Review by farnabaz — APPROVED** [review]
LGTM 👍

**Pull request merged**